### PR TITLE
CL: Add Client support for migration proposals, wire it to gov module

### PR DESCRIPTION
**Pull request by mattverse**
Closes: https://github.com/osmosis-labs/osmosis/issues/4443

## What is the purpose of the change
Adds Client support for migration proposals and wires it to the gov module in app.go

## Brief Changelog
Add Client support for migration proposals for Concentrated Liquidity

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/gamm/client/cli/tx.go [hunk 1]
```diff
@@ -178,6 +181,115 @@ func NewStableSwapAdjustScalingFactorsCmd() *cobra.Command {
...
+	}
+
+	cmd.Flags().String(govcli.FlagTitle, "", "title of proposal")
+	cmd.Flags().String(govcli.FlagDescription, "", "description of proposal")
+	cmd.Flags().String(govcli.FlagDeposit, "", "deposit of proposal")
+	cmd.Flags().String(FlagMigrationRecords, "", "The migration records array")
+
+	return cmd
+}
+
```
Comment (czarcas7ic) [inline x/gamm/client/cli/tx.go:292]: Why are we not using osmocli for these?

Comment (mattverse) [reply x/gamm/client/cli/tx.go:292]: @czarcas7ic  Can osmocli handle specific flags and logics inside the RunE function as well?

Comment (p0mvn) [reply x/gamm/client/cli/tx.go:292]: It should be possible to add custom flag parsers:
https://github.com/osmosis-labs/osmosis/blob/9743eaae0925eed56ff5e7305935a54266220c1c/x/poolmanager/client/cli/tx.go#L43-L45

**Review by p0mvn — COMMENTED** [review]
LGTM

Would like to know if osmocli can be integrated before approving

**Comment by mattverse** [comment]
@czarcas7ic  @p0mvn changed to use osmocli and tried figuring out as much as I could referencing other code, but I have some questions:
1. how do I convert `osmocli.TxCliDesc` object to cobra.Command object? The [govClient Proposal handler](https://github.com/osmosis-labs/osmosis/blob/aad013223a/x/gamm/client/proposal_handler.go#L11) takes in cobra.Command, so we would still need a way to use cobra.Command object.
2. Do flags get automatically mapped to fields? Ex) https://github.com/osmosis-labs/osmosis/blob/aad013223afcd934a7cd84fad24a1f1103a96b34/x/gamm/client/cli/flags.go#L125 Here I am setting a flag for deposit, and in the message field, we have a field named "Initial Deposit" How do I map the flag to the actual field value?
3. How do I manually set a field value for the message? In `govtypes.MsgSubmitProposal`, we have a field named `IsExpedited`, where previously we would just set the field to false and then broadcast the tx. How do I achieve the same goal using osmo cli?

File: x/gamm/client/cli/tx.go (hunk 1 above)
Comment (mattverse) [reply x/gamm/client/cli/tx.go:292]: Resolving this due to issues mentioned below

File: x/gamm/client/cli/tx.go [hunk 2]
```diff
@@ -178,6 +181,115 @@ func NewStableSwapAdjustScalingFactorsCmd() *cobra.Command {
...
+// NewCmdSubmitUpdateMigrationRecordsProposal implements a command handler for replace migration records proposal
```
Comment (czarcas7ic) [inline x/gamm/client/cli/tx.go:239]:
```suggestion
// NewCmdSubmitUpdateMigrationRecordsProposal implements a command handler for update migration records proposal
```

**Review by czarcas7ic — APPROVED** [review]
This looks good, can you add a link here to the issue created for changing this to osmocli before we merge?

**Review by p0mvn — APPROVED** [review]
LGTM

File: x/gamm/client/cli/tx.go (hunk 2 above)
Comment (mattverse) [reply x/gamm/client/cli/tx.go]: @czarcas7ic thanks for flagging! Here is the issue that I have created to track this https://github.com/osmosis-labs/osmosis/issues/4672

**Pull request merged**